cmd/gb-vendor/fetch.go

```diff
@@ -56,7 +56,11 @@ Flags:
 			return err
 		}
 
-		wc, err := repo.Checkout(branch, revision)
+		if m.HasDependencyWithUrl(repo.URL()) {
+			return fmt.Errorf("specified repository is already downloaded")
```

Comment:
I think this needs more information, for example if we've vendored

example.org/foo/bar/baz

then example.org/foo/bar will be the url

so venforing example.org/foo/bar/quux

will fail, but the message is confusing because we haven't vendored this url before, but a sibling of it.